### ISSUE TITLE
Detect wrong use of string equality comparison operator

### ISSUE THREAD

**Issue by sanmai-NL** [koalaman/shellcheck#686]
In GNU bash 4.3.42(1)
```bash
test 's' == 's'
printf '%d\n' "$?"
```
```
0
```

In `dash` 0.5.8
```sh
test 's' == 's'    
```
```
dash: 1: test: s: unexpected operator
```

`==` should not be used as string equality comparison operator since it is not portable.

**Comment by hartwork** [comment]
Any news on getting that check added?

**Comment by koalaman** [comment]
No updates yet. At some point I want to add all the `[` checks for `test`, but since people make quite different assumptions about them, there were too many false positives to just reuse the same code.
